spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/IndicatorCol.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import scala.reflect.ClassTag
+
+/**
+ * Indicator operation represents multi-hot representation of given Tensor.
+ *
+ * The Input Tensor only support Sparse Tensor.
+ * And used to transform the output tensor of CategoricalCol* ops.
+ *
+ * The output tensor should be a DenseTensor with shape (batch, feaLen).
+ *
+ * For example, A input SparseTensor as follows:
+ *  indices(0) = Array(0, 0, 1, 2, 2)
+ *  indices(1) = Array(0, 3, 1, 1, 2)
+ *  values     = Array(1, 2, 2, 3, 3)
+ *  shape      = Array(3, 4)
+ *
+ *  the output tensor should be an 2D 3x4 DenseTensor with isCount = true
```

Comment:
I think here is a little misleading. Should we mention 3 x 4 result tensor is because we set feaLen = 4? BTw, in this case, what if we set feaLen = 2(some value cannot be mapped to multi-hot feature space)

Reply:
well, this Operation is used after `CategoricalCol*` Ops. So the **setValue of feaLen** should **avoid** the case you mentioned. For example, if the Operation used after the `CategoricalColHashBucket`, then the **feaLen** should be set *bucketSize*. This is our solution by far. Looking forward your better advice and thanks.

Reply:
what will happen if the user sets an incorrect value? For example, if the operation is used after the ```CategoricalColHashBucket```, and the user set the ```feaLen``` to bucketSize + 1 or bucketSize -1

Reply:
@yiheng well, if the user sets an incorrect value, it will not work. a alternative solution is that we can get the `feaLen` from the `input.size`, but we should ensure the size of input tensor correctly, which means the input from `CategoricalColHashBucket` should be a SparseTensor with shape(batch_size,**bucketSize**). Then the config parameter `feaLen` can be cancelled. So what's your opinion? Thanks.

Reply:
I think we can let user set the ```feaLen```.

If the ```feaLen``` > ```bucketSize```, looks like it won't affect the multi-hot representation, right? (just the extra bits of the multi-hot vector are always zero)

if the ```feaLen``` < ```bucketSize```, looks like exception will be thrown at line 71 or line 74. Should we check if the tempArr(j) > feaLen before set the value, and throw a more meaningfull error message there?

Reply:
@yiheng  that's a good idea and the **validation** of parameter `feaLen`, pelase check it and thanks.